lib/project_types/extension/models/specification_handlers/checkout_ui_extension.rb

```diff
@@ -62,6 +62,14 @@ def localization(context)
           end
         end
 
+        def read_locale_file(filename)
+          content = File.read(filename, mode: "rt", encoding: "bom|utf-8").strip
+          raise_invalid_encoding_error(filename) unless content.valid_encoding?
+          Base64.strict_encode64(content)
+        rescue ArgumentError
```

Comment:
Would `ArgumentError` be raised if the content is invalid and cannot be encoded? And should we have a test for this if that's the case?

Reply:
Looking at the Base64 code, I think the [strict_encode64](https://github.com/ruby/base64/blob/master/lib/base64.rb#L65) method doesn't raise an ArgumentError (as `strict_decode64` does). As the encoding is validated before, maybe this rescue is unnecessary.